### ISSUE TITLE
How do I extend retention period of existing symbol indices (without symbol packages)?

### ISSUE THREAD

**Issue by dagood** [dotnet/arcade#2415]
At release time, expiration date needs to be [extended from 180 days to 10950 days (30 years)](https://github.com/dotnet/core-eng/issues/3382). In earlier infra, this was done by publishing the same symbol packages again with the longer expiration date.

How is this done when some bits aren't in the symbol packages?

I imagine you *could* by downloading all the bits and pushing them back up, but how is that done in a release context? Is there tooling?

**Comment by jcagme** [comment]
FYI @JohnTortugo

**Comment by tmat** [comment]
I believe we should set the expiration data to the required value right away for all builds, so that we do not need to ever extend it.

**Comment by markwilkie** [comment]
Moving to P7

**Comment by JohnTortugo** [comment]
Anyone see an issue with just extending the expiration data from the start? Who can confirm that the expiration period should be?

**Comment by dagood** [comment]
> Anyone see an issue with just extending the expiration data from the start?

@mikem8361 might have thoughts here. I don't know who would specifically care about overusing symbol indexing/storage space. Maybe until we get a bill, nobody cares. 😄

**Comment by mikem8361** [comment]
If we are talking about extending the expiration date to 30 years for every daily build that might add up and actually cost some storage space. I'm not basing that on any data I have, just a feeling :).

If we can extend only for the potential release builds, then that shouldn't be a problem at all. The old prodcon final publish job had a build variable that could be set to 10950 days.  Not sure how things work for releases anymore. 

We only have republishing as far as tooling to extend the expiration date.

**Comment by jcagme** [comment]
@JohnTortugo is this done? If not should this be done be P7?

**Comment by JohnTortugo** [comment]
It's not done.  [Currently the expiration date is 10 years](https://github.com/dotnet/arcade/blob/6cdfd1ea382ef8566471e3ccaa803d8d83c7f757/eng/common/PublishToSymbolServers.proj#L38). Not a blocker for P6 and IMO not a blocker for P7 as well.

I think now that we are going to have a set or rings where we promote builds through, we might solve this problem by increment the expiration date once the build reach a certain ring. That way we wouldn't increment the expiration date for every build.

**Comment by riarenas** [comment]
@johntortugo should we move this to the Ringed releases epic? https://github.com/dotnet/arcade/issues/2891

**Comment by JohnTortugo** [comment]
Yeah, I think this should be part of the Ringed release epic. Can you please move it? I don't know why but the 'Epic dropdown' isn't showing up for me.

**Comment by riarenas** [comment]
Yep I think it's broken right now 😢

**Comment by dagood** [comment]
Works in ZenHub app.

**Comment by riarenas** [comment]
> Works in ZenHub app.

Didn't even know that was a thing. Thanks Davis

**Comment by dagood** [comment]
I ended up there accidentally at some point. Funny that it's ended up useful. 😄

**Comment by adiaaida** [comment]
@riarenas says that build's symbol retention is now 10 years. Is that long enough? Ricardo, please let me know if that's incorrect

**Comment by dagood** [comment]
I haven't heard anything about the requirement dropping from 30 years to 10 years. Some teams have been ok with 10, but AFAIK the runtime was always 30 years.

**Comment by adiaaida** [comment]
Out of scope for #6418.
